tests/jballerina-unit-test/src/test/resources/test-src/statements/matchstmt/mapping-match-pattern.bal

```diff
@@ -0,0 +1,516 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+function mappingMatchPattern1(any v) returns string {
+    match v {
+        {a: "str"} => {
+            return "match1";
+        }
+        {a: 1} => {
+            return "match2";
+        }
+        {a: true} => {
+            return "match3";
+        }
+    }
+    return "No match";
+}
+
+function testMappingMatchPattern1() {
+    assertEquals("match1", mappingMatchPattern1({a: "str"}));
+    assertEquals("match1", mappingMatchPattern1({a: "str", b: "D"}));
+    assertEquals("match2", mappingMatchPattern1({a: 1}));
+    assertEquals("match3", mappingMatchPattern1({a: true}));
+    assertEquals("No match", mappingMatchPattern1({b: true}));
+}
+
+function mappingMatchPattern2(any v) returns string {
+    match v {
+        {a: "str1", b: "str2"} => {
+            return "match1";
+        }
+        {a: 1, b: 2} => {
+            return "match2";
+        }
+        {a: true, b: false, c: 3} => {
+            return "match3";
+        }
+    }
+
+    return "No match";
+}
+
+function testMappingMatchPattern2() {
+    assertEquals("match1", mappingMatchPattern2({a: "str1", b: "str2"}));
+    assertEquals("match2", mappingMatchPattern2({a: 1, b: 2}));
+    assertEquals("match3", mappingMatchPattern2({a: true, b: false, c: 3}));
+    assertEquals("match2", mappingMatchPattern2({a: 1, b: 2, c: 4}));
+    assertEquals("No match", mappingMatchPattern2({b: true}));
+}
+
+function mappingMatchPattern3(any v) returns string {
+    match v {
+        {a: "str1", b: "str2"} | {a: "str3", b: "str4"} | {a: "str5", b: "str6", c: "str7"}  => {
+            return "match1";
+        }
+        {a: 1, b: 2} | {a: 3, b: 4} | {a: 5, b: 6, c: 7} => {
```

Comment:
The third pattern has an extra variable, right? Shouldn't this fail because "All the match-patterns in a given match-pattern-list must bind the same set of variables."?

Reply:
As discussed offline, I guess we don't have to consider these as variables.